locksmith/src/middlewares/signatureValidationMiddleware.ts

```diff
@@ -0,0 +1,70 @@
+import sigUtil from 'eth-sig-util'
+import ethJsUtil from 'ethereumjs-util'
+import Base64 from '../utils/base64'
+import { Request, Response } from 'express-serve-static-core'
+
+namespace SignatureValidationMiddleware {
+  function extractToken(req: Request) : String| null {
+    if (
+      req.headers.authorization &&
+      req.headers.authorization.split(' ')[0] === 'Bearer'
+    ) {
+      return req.headers.authorization.split(' ')[1]
+    } else {
+      return null
+    }
+  }
+
+  function validatePayload(payload: any, signee: string): Boolean{
+    if (payload.lock) {
+      return (
+        ethJsUtil.toChecksumAddress(payload.lock.owner) ===
+        ethJsUtil.toChecksumAddress(signee)
+      )
+    } else {
+      return false
+    }
+  }
+  function validatePayloadContent(payload: any): Boolean{
+    if (payload.lock) {
+      if (payload.lock.name && payload.lock.owner && payload.lock.address) {
+        return true
+      }
+    }
+    return false
+  }
+
+  export function process(req: any, res: Response, next: any) {
```

Comment:
can we use `next: Function` or whatever the typescripty way to do this is?

Reply:
I think `next` functions in Express never return anything, so you could do `next: () => void`

Reply:
yeah the reason I held off on setting this is in the case that somewhere in the chain of middleware there is something that returns disaster ensues

Reply:
It shouldn't cause any concerns at runtime, so as long as it builds you'll be fine